### PR TITLE
ffms2 formula

### PR DESCRIPTION
**Pull request by kodabb**
This PR is the follow-up of https://github.com/Homebrew/homebrew/pull/31507 with an updated version and fully audited formula.

### CODE REVIEW

File: Library/Formula/ffms2.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class Ffms2 < Formula
+  homepage "https://github.com/FFMS/ffms2"
+  url "https://github.com/FFMS/ffms2/archive/2.20.tar.gz"
+  sha1 "bf09d8ed54e5aad86b352b4d335d29da016f2014"
```
Comment (DomT4) [inline Library/Formula/ffms2.rb]: This should be ` SHA256 `.

File: Library/Formula/ffms2.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  depends_on "pkg-config" => :build
+  depends_on "ffmpeg"
+
+  # needed for the test to exit cleanly
```
Comment (DomT4) [inline Library/Formula/ffms2.rb]: As in the ` test do ` block or?

Comment (mikemcquaid) [inline Library/Formula/ffms2.rb]: Can you link to an upstream issue/PR if there is one?

Comment (kodabb) [inline Library/Formula/ffms2.rb]: Yes it's needed for the `test` block. The patch in the `patch` section is actually my commit which adds this functionality, so no pr or issues are open. It won't be necessary for next version

Comment (mikemcquaid) [inline Library/Formula/ffms2.rb]: Ah ok, you are part of this project? I'm wondering whether it's better for you to maintain this in a tap as we generally frown a bit on self-submissions.

**Comment by kodabb** [comment]
updated formula with sha256 and expanded comment for the patch

File: Library/Formula/ffms2.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  def install
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--enable-avresample",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
+  test do
+    system "ffmsindex"
```
Comment (tdsmith) [inline Library/Formula/ffms2.rb]: Would it make sense to grab a small file from http://samples.mplayerhq.hu/ and try to index it? You can set it up as a `resource` and then run `resource("testfile").stage testpath` to download it to the current directory for the test.

Comment (kodabb) [inline Library/Formula/ffms2.rb]: ok, but for a full test I'd need to validate the output index file with a reference

Comment (mikemcquaid) [inline Library/Formula/ffms2.rb]: I think just checking it doesn't crash would be OK.

Comment (kodabb) [inline Library/Formula/ffms2.rb]: ok I'll get to it

**Comment by kodabb** [comment]
formula updated

**Comment by mikemcquaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format for simple version updates is `ffms2 2.20 (new formula)`.

**Pull request closed without merging**